doc/service-integrations.md

```diff
@@ -0,0 +1,75 @@
+# Integration with upstream services
+
+## Overview
+
+In a nutshell, the Shields Badge Server handles the responsibilities of accepting requests for badges, and then serving those badges back to users.
+
+A grossly oversimplified visualization would probably look something like this:
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant B as Badge Server
+    participant P as Data Provider
+    User->>+B: I'd like a badge for my project please!
+    B->>+P: Get data for project
+    P-->>-B: Data
+    B->>B: Make badge with data
+    B-->>-User: Here's your badge!
+```
+
+Shields is not a system of record (we're not the package registry, pipeline tool, etc.) so when Shields receives a request for a badge, the badge server will first have to reach out to the system of record in order to get the data points it needs to create your badge.
+
+For example, if you ask Shields for a build status badge for your CircleCI pipeline, then Shields has to reach out to CircleCI to figure out what the status of your pipeline is (CircleCI would be the "Data Provider" actor in the above diagram). Similarly if you want a badge that shows the count of downloads of your npm package, Shields has to reach out to the npm.js registry.
+
+That covers the gist, but the actual story is a bit more involved and complicated than that of course. There's a number of other components along the way, ranging from the browser on your local machine to extra services and actors we have deployed as part of the Shields.io runtime ecosystem which help ensure we can provide a stable and reliable deployment of the badge server-as-a-service. Additionally, badges rendered in GitHub have some additional factors at play that impose some additional constraints, detailed in the next section.
+
+### GitHub Badge Rendering
+
+A common usage pattern for badges is to embed them in your project's README files so relevant information is conveyed to the project's users. This means badges are often utilized and rendered in source control management platforms, like GitHub.
+
+[GitHub utilizes a proxy service, called camo][camo], for handling the images that you see when you browse project pages on github.com, and this is utilized for badges too (both svg and png formats). GitHub does this for a number of reasons, including to anonymize requests and protect your privacy. However, this also requires the upstream images (including badges) to be returned quickly in order for those images to show up on your screen, with a rough ceiling of 3-4 seconds. If the upstream image provider is too slow to respond, then camo will timeout and the image won't be displayed.
```

Comment:
Worth mentioning here that also some other places badges are commonly viewed do this too. For example PyPI also runs a camo proxy for images and it is one of our biggest sources of traffic

Also it feels like the concept of other people cacheing our badges (with Camo) and us cacheing our badges (with CloudFlare) kind of go together in that they happen in basically the same phase of the "flow" described by the diagram. So maybe it makes sense to talk about those things in the same section or adjacent sections